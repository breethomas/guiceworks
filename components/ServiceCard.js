// @flow
import React from 'react'
import css, { hover, media2, media3, media4 } from '../styles/css'
import Icon from './Icons'
import Link from './Link'
import type { ServiceProps } from '../types/app.js.flow'

type Props = {
  service: ServiceProps,
}

const style = css(
  {
    display: 'flex',
    flexFlow: 'column wrap',
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    height: '100vw',
    maxHeight: 350,
    textDecoration: 'none',
    backgroundColor: 'rgba(255, 255, 255, 0)',
    transition: 'background-color 0.4s',
  },
  hover({ backgroundColor: 'rgba(0, 0, 0, 0.1)' }),
  media2({ width: '50%' }),
  media3({ width: '33.33333%' }),
  media4({ width: '25%' }),
)

const textStyle = css({
  marginTop: 40,
  fontSize: 14,
  textTransform: 'uppercase',
})

export default ({ service, ...props }: Props) => (
  <Link
    className={style}
    as={`/services/${service.id}`}
    href={`/services?id=${service.id}`}
    key={service.title}
    {...props}
  >
    <Icon kind={service.icon.replace('.svg', '')} size={48} />
    <span className={textStyle}>{service.title}</span>
  </Link>
)
