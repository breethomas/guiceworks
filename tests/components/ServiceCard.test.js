import React from 'react'
import renderer from 'react-test-renderer'
import ServiceCard from '../../components/ServiceCard'

describe('ServiceCard', () => {
  let component

  beforeEach(() => {
    const service = {
      body: 'Service 1 _body_',
      icon: 'icon.svg',
      id: 'service-1',
      title: 'Service 1 title',
    }

    component = (
      <ServiceCard
        service={service}
      />
    )
  })

  it('shows the correct snapshot for the ServiceCard component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
