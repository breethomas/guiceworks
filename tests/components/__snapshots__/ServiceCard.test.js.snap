// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCard shows the correct snapshot for the ServiceCard component tree 1`] = `
<a
  className={
    Object {
      "data-css-1c6omyw": "",
    }
  }
  href="/services/service-1"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    height={48}
    viewBox="0 0 32 32"
    width={48}
  >
    <g
      fill="currentColor"
    >
      <path
        d="M29.6104,7.5859 C28.3044,5.4789 26.5234,3.6969 24.4154,2.3909 L19.0084,7.8239 C19.0174,7.8269 19.0264,7.8309 19.0354,7.8339 L11.8144,15.0549 L16.9694,20.2099 L24.1794,13.0009 C24.1804,13.0039 24.1824,13.0079 24.1834,13.0119 L29.6104,7.5859 Z"
      />
      <path
        d="M24.668,15.1348 C24.695,15.4198 24.71,15.7078 24.71,15.9998 C24.71,20.8108 20.811,24.7098 16,24.7098 C11.189,24.7098 7.29,20.8108 7.29,15.9998 C7.29,11.1898 11.189,7.2898 16,7.2898 L16,-0.0002 C7.164,-0.0002 0,7.1638 0,15.9998 C0,24.8368 7.164,31.9998 16,31.9998 C24.837,31.9998 32,24.8368 32,15.9998 C32,13.5988 31.471,11.3228 30.523,9.2798 L24.668,15.1348 Z"
      />
    </g>
  </svg>
  <span
    className={
      Object {
        "data-css-d8adml": "",
      }
    }
  >
    Service 1 title
  </span>
</a>
`;
